### PR TITLE
feat(e2e): make testenv pluginable

### PR DESCRIPTION
**Pull request by chrisgacsal**
## Description

Add `testenv` pluginable in order to support running end-to-end test suite on platforms supported by VMClarity.

## Type of Change

[ ] Bug Fix  
[x] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Review by pbalogh-sa — DISMISSED** [review]
Thx, LGTM

File: e2e/testenv/testenv_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		t.Run(test.Name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+
+			os.Clearenv()
+			for k, v := range test.EnvVars {
+				err := os.Setenv(k, v)
+				g.Expect(err).Should(Not(HaveOccurred()))
+			}
+
+			config, err := NewConfig()
```
Comment (paralta) [inline e2e/testenv/testenv_test.go:72]: I think there is a missing error check here, maybe use `ExpectedNewErrorMatcher`

File: e2e/suite_test.go [hunk 2]
```diff
@@ -19,67 +19,64 @@ import (
...
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	if !reuseEnv {
+		ginkgo.By("setup test environment")
+		err = testEnv.SetUp(ctx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
+		ginkgo.By("starting test environment")
+		err = testEnv.Start(ctx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
```
Comment (paralta) [inline e2e/suite_test.go:70]: We could log "reusing existing environment" when `reuseEnv` true

Comment (paralta) [reply e2e/suite_test.go:70]: and we can use `config.Reuse` directly here

**Review by paralta — APPROVED** [review]
LGTM, just some small suggestions :D

File: e2e/suite_test.go (hunk 2 above)
Comment (chrisgacsal) [reply e2e/suite_test.go:70]: We would need to store config instead. But will make the change.

File: e2e/testenv/testenv_test.go (hunk 1 above)
Comment (chrisgacsal) [reply e2e/testenv/testenv_test.go:72]: It does not, it is in the next line here: https://github.com/openclarity/vmclarity/pull/773/files#diff-613134a90e21026ae742ce95d821bbd0042fcd8e47caf2e6c7a5ae3eb45d54e4R74

File: e2e/suite_test.go (hunk 2 above)
Comment (chrisgacsal) [reply e2e/suite_test.go:70]: Both fixed.

**Review by paralta — APPROVED** [review]

**Pull request merged**